### DISCUSSION TITLE
REST API Reference on Docker Container

### DISCUSSION THREAD

**Discussion by rackrick** [Keyfactor/ejbca-ce#661] [Q&A] [answered]
Hi there,

I setup a EJBCA Instance using the docker container. Now I want to use the rest api the issue some certs.
To do that I was looking for a API reference but the one mentioned in the documentation returns a 404 not found.

https://doc.primekey.com/ejbca/ejbca-operations/ejbca-ca-concept-guide/protocols/ejbca-rest-interface#EJBCARESTInterface-RESTAPIReferencerest_api_reference

I found an old post with the same issue but no answer:
https://github.com/Keyfactor/ejbca-ce/discussions/177

Is there any way to get the api reference without the need to setup a instance without docker or to do a build without the prod flag?

**Comment by primetomas** [comment]
You mean this URL right? https://localhost:8442/ejbca/doc/rest

**Reply by rackrick** [reply]
correct...
since its a docker container I can't use localhost and tried the ip instead.
Also I noticed the port in the screenshot differs from the one mentioned in the documentation. So I assumed `8443` would be correct.

**Comment by primetomas** [comment, marked as answer]
You can download swagger.json from GitHub: https://github.com/Keyfactor/ejbca-ce/blob/main/modules/ejbca-rest-api/resources/swagger.json

Then you can run swagger locally by for exampel:
docker run -p 80:8080 -e SWAGGER_JSON=/foo/swagger.json -v /tmp:/foo swaggerapi/swagger-ui

That gives you Swagger API.

**Comment by rackrick** [comment]
Thanks for your help... I didn't thinkt about the swagger.json. That helps... you can even open it on https://editor.swagger.io/ to keep it simple without the container.

**Reply by primetomas** [reply]
Ah, thanks for that. I'll add a note about this in the documentation for the next release.
